velox/dwio/dwrf/reader/SelectiveStructColumnReader.cpp

```diff
@@ -193,13 +193,43 @@ void SelectiveStructColumnReader::read(
   readOffset_ = offset + rows.back() + 1;
 }
 
+namespace {
+void fillRowVectorChildren(
+    memory::MemoryPool& pool,
+    const RowType& rowType,
+    std::vector<VectorPtr>& children) {
+  for (auto i = 0; i < children.size(); ++i) {
+    auto type = rowType.childAt(i);
+    if (type->kind() == TypeKind::ROW) {
+      std::vector<VectorPtr> innerChildren(type->size());
+      fillRowVectorChildren(pool, type->as<TypeKind::ROW>(), innerChildren);
+      children[i] = std::make_shared<RowVector>(
+          &pool, type, BufferPtr(nullptr), 0, std::move(innerChildren));
+    }
+  }
+}
+} // namespace
+
 void SelectiveStructColumnReader::getValues(RowSet rows, VectorPtr* result) {
   assert(!children_.empty());
   VELOX_CHECK(
       *result != nullptr,
       "SelectiveStructColumnReader expects a non-null result");
-  RowVector* resultRow = dynamic_cast<RowVector*>(result->get());
+  auto resultRow = dynamic_cast<RowVector*>(result->get());
```

Comment:
nit: this can be a static cast if the check is moved ahead of this and changed to result->type()->isRow()